src/Symfony/Component/Notifier/Bridge/Clickatell/Tests/ClickatellTransportTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Symfony\Component\Notifier\Bridge\Clickatell\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpClient\MockHttpClient;
+use Symfony\Component\Notifier\Bridge\Clickatell\ClickatellTransport;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+class ClickatellTransportTest extends TestCase
+{
+    public function testToString()
+    {
+        $transport = new ClickatellTransport('authToken', 'fromValue', $this->createMock(HttpClientInterface::class));
+        $transport->setHost('clickatellHost');
+
+        $this->assertSame('clickatell://clickatellHost?from=fromValue', (string) $transport);
+    }
+
+    public function testSupports()
+    {
+        $transport = new ClickatellTransport('authToken', 'fromValue', $this->createMock(HttpClientInterface::class));
+
+        $this->assertTrue($transport->supports(new SmsMessage('+33612345678', 'testSmsMessage')));
+        $this->assertFalse($transport->supports($this->createMock(MessageInterface::class)));
+    }
+
+    public function testExceptionIsThrownWhenNonMessageIsSend()
+    {
+        $transport = new ClickatellTransport('authToken', 'fromValue', $this->createMock(HttpClientInterface::class));
+        $transport->send($this->createMock(MessageInterface::class));
+
+        $this->expectException(LogicException::class);
+    }
+
+    public function testExceptionIsThrownWhenHttpSendFailed()
+    {
+        $response = $this->createMock(ResponseInterface::class);
+        $response->expects($this->exactly(2))
+            ->method('getStatusCode')
+            ->willReturn(500);
+        $response->expects($this->once())
+            ->method('getContent')
+            ->willReturn(json_encode([
+                'error' => [
+                    'code' => 105,
+                    'description' => 'Invalid Account Reference EX0000000',
+                    'documentation' => 'https://documentation-page',
+                ],
+            ]));
+
+        $client = new MockHttpClient($response);
+
+        $transport = new ClickatellTransport('authToken', 'fromValue', $client);
+        $transport->send(new SmsMessage('+33612345678', 'testSmsMessage'));
+
+        $this->expectException(TransportException::class);
+        $this->expectExceptionMessage('Unable to send SMS with Clickatell: Error code 105 with message "Invalid Account Reference EX0000000" (https://documentation-page).');
```

Comment:
```suggestion
        $this->expectException(TransportException::class);
        $this->expectExceptionMessage('Unable to send SMS with Clickatell: Error code 105 with message "Invalid Account Reference EX0000000" (https://documentation-page).');

        $transport->send(new SmsMessage('+33612345678', 'testSmsMessage'));
```

Reply:
I'm sorry but I don't understand.

In my first commit `$this->expectException` was in the top of the function and a suggestion was to down that (https://github.com/symfony/symfony/pull/38922#discussion_r538348743)

And now, this suggestion is to top that :thinking: 

There is a rule for that ?

Reply:
Sure I will explain, IIRC the expectation was at the top of the **method**, now it is on the bottom of the **method**.

My rule is, that I place an expectation for an exception + exception message right before the method call which should throw this exception. By doing this (mostly when not expecting a message and only and exception class), I am quite sure that it is exactly this exception and not lets say raised by a class which couldn't be instantiated or something else. I hope I made it more clear now.

Reply:
That's true, often I place my expectation at the top of the function, but you are right, an other exception not expected with the same class and the same message can be throwed and make the test useless.

Thank you for the tips, I will apply that for the next time :+1: